src/frontend/magnify/src/components/design-system/Fieldset/Fieldset.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { Box, Text } from 'grommet';
+import { MarginType } from 'grommet/utils';
+
+export interface FieldsetProps {
+  label: string;
+  name: string;
+  children: React.ReactNode;
+  margin?: MarginType;
+  required?: boolean;
+  displayErrors?: boolean;
+  errors?: string[];
+}
+
+const Fieldset = ({
+  name,
+  label,
+  children,
+  margin = { vertical: 'small' },
+  required = false,
+  displayErrors = false,
+  errors = [],
+}: FieldsetProps) => {
+  return (
+    <Box
+      border={{ side: 'all', color: 'brand' }}
+      round="8px"
+      pad={{ vertical: 'xsmall', horizontal: 'small' }}
+      margin={margin}
+    >
+      <Box margin={{ left: '11px' }}>
+        <label htmlFor={name}>
+          <Text size="xsmall" weight="bold" color="brand">
```

Comment:
Maybe the label should be also red if required ?
```suggestion
          <Text size="xsmall" weight="bold" color={required ? 'status-error' : 'brand'}>
```

Reply:
Isn't it too violent ? I see the red start on a lot of website, but the whole label... I don't know...

Reply:
Maybe red is too `errorish` but I think the label should have the same color than the appended `*` from lines 37-39
Maybe use `status-warning` or something else for both ?

Reply:
I don't think I saw this a lot before. For the sake of the demonstration, here are a dummy form with required and not required fields with each option. I'm not very comfortable with last 2 options, but maybe we can ask the designer ? What do you think ?

![Screenshot from 2022-07-08 09-22-53](https://user-images.githubusercontent.com/25184106/177939938-8a89ed7e-60f3-4a9c-b071-638972873ff0.png)
![Screenshot from 2022-07-08 09-23-32](https://user-images.githubusercontent.com/25184106/177939940-7c559349-e85b-4feb-aeae-19228f62009f.png)
![Screenshot from 2022-07-08 09-23-50](https://user-images.githubusercontent.com/25184106/177939942-9420a0c0-be83-4a81-b7d1-8bfc95d9fbe6.png)

Reply:
This was just a proposal, let's keep the label on his initial color :+1:

Reply:
Thanks for taking time to do some tries :)